ocaml/forkexecd/helper/vfork_helper.c

```diff
@@ -335,14 +335,44 @@ reset_signal_handlers(void)
 static void
 clear_cgroup(void)
 {
-    int fd = open("/sys/fs/cgroup/systemd/cgroup.procs", O_WRONLY|O_CLOEXEC);
-    if (fd >= 0) {
-        char string_pid[32];
-        int ignored __attribute__((unused));
-        sprintf(string_pid, "%d\n", (int) getpid());
-        ignored = write(fd, string_pid, strlen(string_pid));
+    // list of files to try, terminated by NULL
+    static const char *cgroup_files[] = {
+        "/sys/fs/cgroup/systemd/cgroup.procs",
+        "/sys/fs/cgroup/cgroup.procs",
+        NULL
+    };
+
+    char string_pid[32];
+    int last_error = 0;
+    const char *last_error_msg = NULL;
+    const char *last_fn = NULL;
+
+    snprintf(string_pid, sizeof(string_pid), "%d\n", (int) getpid());
```

Comment:
https://pubs.opengroup.org/onlinepubs/9799919799/basedefs/sys_types.h.html says "The implementation shall support one or more programming environments in which the widths of blksize_t, pid_t, size_t, ssize_t, and suseconds_t are no greater than the width of type long", so perhaps "%ld" and `(long)` might be more appropriate.

Reply:
For compatibility in this case `pid_t` specifically cannot be bigger than in `int`. But won't hurt.